### PR TITLE
[10.0] Migrate printer_zpl2 module

### PR DESCRIPTION
**Pull request by sla-subteno-it**

### CODE REVIEW

**Comment by lasley** [comment]
Thanks for the contribution @alnslang. #68 already had the base_report_to_printer upgraded, so I went ahead and merged it. Can you please rebase onto the current 10.0 so that this PR becomes exclusive to `printer_zpl2`?

**Comment by sla-subteno-it** [comment]
@lasley I rebase my branch and remove my modifications about base_report_to_printer. Now, my PR becomes exclusive to printer_zpl2.

**Comment by lasley** [comment]
@alnslang - Looks like tests are failing due to a missing dependency. Please create a `requirements.txt` in the root directory & add `zpl2` as a line.

Looks like this is done incorrectly on the other branches; I'll submit a PR to fix those.

**Comment by sla-subteno-it** [comment]
Thanks @lasley for your help. I just adding requirements.txt in root directory

File: printer_zpl2/__manifest__.py
Comment (lasley) [inline printer_zpl2/__manifest__.py]: 10.0.1.0.0

File: printer_zpl2/__manifest__.py
Comment (lasley) [inline printer_zpl2/__manifest__.py]: s/external_dependancies/external_dependencies

**Comment by blutecsolutions** [comment]
Greetings !
What's the status of this migration ?
Thanks & we appreciate all efforts.

**Comment by Garamotte** [comment]
Hi @blutecsolutions 
The migration should be finished, waiting for reviews to be approved (or fixed).

**Review by lasley — APPROVED** [review]
Few comments, nothing major

File: printer_zpl2/__manifest__.py
Comment (Garamotte) [reply printer_zpl2/__manifest__.py]: This was already changed, I don't know how you commented on that line...

**Comment by Garamotte** [comment]
Typo fixed, good catch @lasley :)

**Comment by blutecsolutions** [comment]
Is it available for download now?

**Comment by blutecsolutions** [comment]
Awaiting with impatience :) Can we have an update please.

**Comment by lasley** [comment]
Instead of just waiting, you could provide the second review we need to merge it.

File: printer_zpl2/views/printing_label_zpl2.xml
Comment (lasley) [inline printer_zpl2/views/printing_label_zpl2.xml]: Please remove the `base_report_to_printer.` from this ID in order to clear up the Runbot warning

```
id="printing_menu_label"
```

Comment (sla-subteno-it) [reply printer_zpl2/views/printing_label_zpl2.xml]: base_report_to_printer. removed.

Thanks

Comment (Garamotte) [reply printer_zpl2/views/printing_label_zpl2.xml]: The goal was to "provide" a unique menu for labels.
If another module wants to provide labels, it will create a new menu (in this case, this "Labels" menu is useless), or need to depend on printer_zpl2...

How could we achieve this ? Add the "Labels" menu in the base_report_to_printer module instead ?

**Comment by blutecsolutions** [comment]
Can we download this module ?

File: printer_zpl2/models/printing_label_zpl2.py
Comment (elicoidal) [inline printer_zpl2/models/printing_label_zpl2.py]: Width

File: printer_zpl2/models/printing_label_zpl2_component.py
Comment (elicoidal) [inline printer_zpl2/models/printing_label_zpl2_component.py:77]: check digit or check sum?

**Review by elicoidal — APPROVED** [review]
small details

File: printer_zpl2/models/printing_label_zpl2_component.py
Comment (Garamotte) [reply printer_zpl2/models/printing_label_zpl2_component.py:77]: It's called check digit in the documentation : https://www.zebra.com/content/dam/zebra/manuals/en-us/software/zpl-zbi2-pm-en.pdf (page 1299).

Comment (elicoidal) [reply printer_zpl2/models/printing_label_zpl2_component.py:77]: :+1:

**Comment by Garamotte** [comment]
@elicoidal Thanks, typo fixed.
I also removed the "Labels" menu, that became useless since last change asked by @lasley and renamed the "ZPL II" menu to "ZPL II Labels".

**Comment by elicoidal** [comment]
@sylvain-garancher Excellent contribution :)

**Comment by Garamotte** [comment]
@lasley @pedrobaeza This PR is ready to merge.

**Pull request merged**
